frontend/packages/dev-console/src/components/edit-deployment/EditDeploymentForm.tsx

```diff
@@ -45,6 +45,10 @@ const EditDeploymentForm: React.FC<FormikProps<FormikValues> & {
   const resourceType = getResourcesType(resource);
   const isNew = !resource.metadata.name || resource.metadata.name === '~new';
 
+  React.useEffect(() => {
+    !isNew && setFieldValue('formData.name', resource.metadata.name);
+  }, [formData.name]);
+
```

Comment:
My understanding of the issue is that ImageStreamTag dropdowns change the deployment name.

This hook now runs when the name is changed and then reset the name (change the name again) to the resource name.

Instead of changing the name back and forward, we should check why it's changed in the first place and is this really needed!

Reply:
@Lucifergene ^^